data_structures/src/chain.rs

```diff
@@ -131,6 +131,9 @@ pub enum BlockError {
     /// The first transaction of the block is no mint.
     #[fail(display = "The block first transaction is not a mint transactions")]
     NoMint,
+    /// The block has not signature
+    #[fail(display = "The block has not signature")]
+    NoSignature,
```

Comment:
This error was useful when signature was an `Option`, now it can never happen: blocks must have a signature.